sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -632,6 +633,24 @@ object ScalaJSPluginInternal {
     }
   }
 
+  protected def javaSysProps2JSLib(javaOptions: Seq[String]): Option[VirtualJSFile] = {
+    val javaSysPropsPattern = "-D([^=]*)=(.*)".r
+    val formattedProps = javaOptions.collect {
+      case javaSysPropsPattern(propName, propValue) =>
+        "\"" + escapeJS(propName) + "\": \"" + escapeJS(propValue) + "\""
+    }
```

Comment:
That is an implementation detail of the sbt plugin. I do not say it must fail here. I just say if you write:

```scala
javaOptions += "-foo"
```

running must fail.

Just like it does on the JVM:

```
> set javaOptions += "-foo"
// chatter
> set fork := true
// chatter
> run
// chatter
[error] Unrecognized option: -foo
[error] Error: Could not create the Java Virtual Machine.
[error] Error: A fatal exception has occurred. Program will exit.
```